todo_app/tests/test_IntUnit.py

```diff
@@ -0,0 +1,41 @@
+import pytest
+from dotenv import load_dotenv, find_dotenv
+from unittest.mock import patch
+import requests
+import os
+from threading import Thread
+from todo_app import app    
+
+@pytest.fixture(scope='module')
+def test_app():
+       application = app.create_app()
+       thread = Thread(target=lambda: application.run(use_reloader=False))
+       thread.daemon = True
+       thread.start()
+       yield app 
+       thread.join(1)
+
+@pytest.fixture
+def client():
+       file_path = find_dotenv('.env.test')
```

Comment:
It would be good to move this loading of dotenv into the `test_app` method, as that's strictly where it's required, and this can be confusing (e.g, in your e2e test, if run from the command line alone it doesn't currently load the .env.test variables in)